react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java

```diff
@@ -579,33 +586,42 @@ public void toggleEditorMode(boolean htmlModeEnabled) {
     }
 
     public void appendUploadMediaFile(final int mediaId, final String mediaUri, final boolean isVideo) {
-       if (isMediaUploadCallbackRegistered() && mMediaPickedByUserOnBlock) {
-           String mediaType = getMediaType(isVideo);
-           mMediaPickedByUserOnBlock = false;
-           List<RNMedia> mediaList = new ArrayList<>();
-           mediaList.add(new Media(mediaId, mediaUri, mediaType));
-           mPendingMediaUploadCallback.onUploadMediaFileSelected(mediaList);
-       } else {
-           // we can assume we're being passed a new image from share intent as there was no selectMedia callback
-           sendOrDeferAppendMediaSignal(mediaId, mediaUri, isVideo);
-       }
+//       if (isMediaUploadCallbackRegistered() && mMediaPickedByUserOnBlock) {
+//           String mediaType = getMediaType(isVideo);
+//           mMediaPickedByUserOnBlock = false;
+//           List<RNMedia> mediaList = new ArrayList<>();
+//           mediaList.add(new Media(mediaId, mediaUri, mediaType));
+//           mPendingMediaUploadCallback.onUploadMediaFileSelected(mediaList);
+//       } else {
+//           // we can assume we're being passed a new image from share intent as there was no selectMedia callback
+//           sendOrDeferAppendMediaSignal(mediaId, mediaUri, isVideo);
+//       }
     }
 
     public void appendUploadMediaFiles(ArrayList<Media> mediaList) {
         if (isMediaUploadCallbackRegistered() && mMediaPickedByUserOnBlock) {
             mMediaPickedByUserOnBlock = false;
             List<RNMedia> rnMediaList = new ArrayList<>();
-            for (Media media : mediaList) {
-                rnMediaList.add(new Media(media.getId(), media.getUrl(), media.getType()));
+
+            if (mAppendsMultipleSelectedToSiblingBlocks && 1 < mediaList.size()) {
+                rnMediaList.add(mediaList.get(0));
+                mPendingMediaUploadCallback.onUploadMediaFileSelected(rnMediaList);
+
+                for (Media mediaToAppend : mediaList.subList(1, mediaList.size())) {
+                    sendOrDeferAppendMediaSignal(mediaToAppend.getId(), mediaToAppend.getUrl(),
+                            mediaToAppend.getType());
+                }
+            } else {
+                rnMediaList.addAll(mediaList);
+                mPendingMediaUploadCallback.onUploadMediaFileSelected(rnMediaList);
             }
-            mPendingMediaUploadCallback.onUploadMediaFileSelected(rnMediaList);
-            mPendingMediaUploadCallback = null;
         }
+
+        mAppendsMultipleSelectedToSiblingBlocks = false;
```

Comment:
Not sure if it is needed since we already set it to the correct value each time we call `requestMediaPickFromMediaLibrary`

Reply:
Really good question ;)! I believe `appendUploadMediaFiles` can be invoked even as a result for `requestMediaPickFromDeviceLibrary`, `requestMediaPickerFromDeviceCamera` etc. 

Imagine the following scenario 
1. _requestMediaPickFrom**MediaLibrary**_ is invoked and `mAppendsMultipleSelectedToSiblingBlocks` is set to `true`
2. `appendUploadMediaFiles` is invoked and we don't re-set `mAppendsMultipleSelectedToSiblingBlocks` back to `false`
3. _requestMediaPickerFrom**DeviceCamera**_ is invoked
4. `appendUploadMediaFiles` is invoked and `mAppendsMultipleSelectedToSiblingBlocks` is `true` even though it should be `false`

I like your thinking and I think we might want to consider removing this line as you suggested and explicitly setting `mAppendsMultipleSelectedToSiblingBlocks` in all the other `requestMediaPickerXYZ` as preparation for the future. What do you think about it @mkevins ? Is there a change we'll ever enable multi-select for the other sources?

Reply:
Very good question! For the image block, the only time multiple selection is possible is via the WordPress Media Library. For this case, the special logic to append blocks has been implemented. In all other cases, I think it should currently be false. IMO this value should only be true for that special case: image block, media library selected.

> Is there a chance we'll ever enable multi-select for the other sources?

I'm not sure about that. I've set the flag to false explicitly in the other cases as well, and that can be updated later if the need arises.

Reply:
Do we need to add it also to `requestMediaPickFrom(..)`  or is it used for a different scenario?

Reply:
Right! Nice catch! Updated.